go/arrow/flight/flight_test.go

```diff
@@ -311,3 +312,65 @@ func TestServer(t *testing.T) {
 		t.Fatalf("got %d, want %d", numRows, fi.TotalRecords)
 	}
 }
+
+type flightMetadataWriterServer struct{}
+
+func (f *flightMetadataWriterServer) DoGet(tkt *flight.Ticket, fs flight.FlightService_DoGetServer) error {
+	recs := arrdata.Records[string(tkt.GetTicket())]
+
+	w := flight.NewRecordWriter(fs, ipc.WithSchema(recs[0].Schema()))
+	defer w.Close()
+	for idx, r := range recs {
+		w.WriteWithAppMetadata(r, []byte(fmt.Sprintf("%d_%s", idx, string(tkt.GetTicket()))))
```

Comment:
nit: not sure go convention here.  But in other langauges I find it useful to document literal like  parameters (the second parameter in this case indicating the second parameter name.

i.e. `/*metadata=*/[]byte(fmt.Sprintf("%d_%s", idx, string(tkt.GetTicket())`

Reply:
Go doesn't particularly have a convention for that type of thing, most of the time it's handled by whatever IDE you're using which would be able to tell you what each argument was by hovering over the function. But it's easy to add some comments such as you described, i'll update

Reply:
So when running `gofmt` after adding the `/*metadata*/` before `[]byte(fmt.Sprintf("%d_%s", idx, string(tkt.GetTicket())))` it insisted on the comment being put after the `r` before the comma, so it appears that the convention for Go is for the comment identifying the literal to be *after* the value, rather than before it. So that's what i'll do.